x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -262,6 +268,36 @@ export function getWaterfall(
     entryWaterfallTransaction
   );
 
+  function isAncestor(
+    waterfallItemsList: IWaterfallItem[],
+    ancestorTransactionId: string,
+    currentItem: IWaterfallError | IWaterfallItem
+  ): boolean {
+    if (!currentItem.parentId) {
+      return false;
+    }
+    if (currentItem.parentId === ancestorTransactionId) {
+      return true;
+    }
+
+    const parentItem: IWaterfallItem | undefined = waterfallItemsList.find(
+      ({ id }) => id === currentItem.parentId
+    );
+
+    if (parentItem) {
+      return isAncestor(waterfallItemsList, ancestorTransactionId, parentItem);
+    }
+    return false;
+  }
+
+  const errorItems: IWaterfallError[] = trace.errorDocs
+    .map(errorDoc =>
+      getErrorItem(errorDoc as APMError, items, entryWaterfallTransaction)
```

Comment:
I think TS will infere that `errrorDoc` is an APMError, I guess you can drop this cast.